src/utils/mod.rs

```diff
@@ -463,19 +463,23 @@ pub(crate) fn user_has_perms_cache(
 pub(crate) fn user_perms(cache: impl AsRef<Cache>, channel_id: ChannelId) -> Result<Permissions> {
     let cache = cache.as_ref();
 
-    let Some(channel) = cache.channel(channel_id) else {
+    let Some(guild_id) = cache.channels.get(&channel_id).map(|c| *c) else {
```

Comment:
I think you meant to write this?:
```suggestion
    let Some(guild_id) = cache.channels.get(&channel_id).map(|c| c.guild_id) else {
```

Reply:
No, cache.channels is map of ChannelId to GuildId. See #2602